cni/cmd/istio-cni/main.go

```diff
@@ -34,9 +34,12 @@ import (
 )
 
 var (
-	nsSetupBinDir          = "/opt/cni/bin"
-	injectAnnotationKey    = annotation.SidecarInject.Name
-	sidecarStatusKey       = annotation.SidecarStatus.Name
+	nsSetupBinDir       = "/opt/cni/bin"
+	injectAnnotationKey = annotation.SidecarInject.Name
+	sidecarStatusKey    = annotation.SidecarStatus.Name
+	// TODO move to API
```

Comment:
@bianpengyuan @stewartbutler PTAL

Reply:
LGTM. I'dd add a test for it in main_test.go and also make sure there are some integration tests for it though.

Reply:
I'd rather avoid adding another annotations just for this. I think there are definitely already some other signals available in the pod spec to tell a proxyless agent? (env var, args ..?)

Reply:
Coupling to the env vars which are generally considered unstable feels a bit sketchy – but I can do that to avoid adding annotations.

Reply:
Yeah we've already been parsing PROXY_CONFIG env var, so long it is covered by integration test I think it is fine, since it is just an implementation detail.